File: ForestManagement/App.DTO/LandProperty/LandPropertyDto.cs
using App.Contracts.Enums;

namespace App.DTO.LandProperty;

/// <summary>
/// Full property details response DTO with related cadasters
/// </summary>
public class LandPropertyDto
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public int RegistrationNumber { get; set; }
    public string County { get; set; } = string.Empty;
    public string Parish { get; set; } = string.Empty;
    public string Village { get; set; } = string.Empty;
    public DateTime? BoughtDate { get; set; }
    public DateTime? SoldDate { get; set; }
    public EPropertyStatus Status { get; set; }
    public Guid CompanyId { get; set; }
    public string CompanyName { get; set; } = string.Empty;
    
    // Related cadasters
    public ICollection<Cadaster.CadasterListDto> Cadasters { get; set; } = new List<Cadaster.CadasterListDto>();
}

File: frontend/src/lib/dtos/land-property/land-property.dto.ts
export type PropertyStatus = 'Active' | 'Inactive' | 'Sold';

export type PropertyCadasterLinkDto = {
	id: string;
	cadastralNumber: string;
};

export type LandPropertyDto = {
	id: string;
	name: string;
	registrationNumber: number;
	county: string;
	parish: string;
	village: string;
	boughtDate: string | null;
	soldDate: string | null;
	status: PropertyStatus | number | string;
	cadastralNumbers?: string[];
	cadasters?: PropertyCadasterLinkDto[];
	companyId: string;
	companyName: string;
	isFsc?: boolean;
};

export type LandPropertyUpdateDto = {
	id: string;
	name: string;
	registrationNumber: number;
	county: string;
	parish: string;
	village: string;
	boughtDate: string | null;
	soldDate: string | null;
	status: PropertyStatus | number;
	companyId: string;
	isFsc?: boolean;
};

export type CadasterLinkDto = {
	id: string;
	cadastralNumber: string;
	forestArea?: number;
	forestStandCount?: number;
};

export type ActivityDto = {
	id: string;
	description: string;
	quantity: number;
	unit: string | null;
	notes: string | null;
	date: string;
	userId: string;
	userName: string;
	activityTypeId: string;
	activityTypeName: string;
	cadasterId: string | null;
	cadasterCadastralNumber: string | null;
	forestStandId: string | null;
	forestStandNumber: number | null;
	landPropertyId: string | null;
	landPropertyName: string | null;
	applicationStatus: number | null;
};
